### PR TITLE
Memory allocation instructions

### PR DESCRIPTION
**Pull request by fgmccabe**
Have added a sketch of two instructions dealing with memory. 

Initial description only.

### CODE REVIEW

File: proposals/interface-types/working-notes/instr.md [hunk 1]
```diff
@@ -95,20 +110,271 @@ is the number of bytes in the string -- not the number of unicode characters.
 These instructions also reference a memory index literal -- which defaults to 0
 -- which indicates which memory the string is held in.
 
-### memory-to-array
+### Memory and Arrays
+
+| | |
+| -- | -- |
+| `memory-to-array` | Lift a region of memory as contents of an array |
+| `array-to-memory` | Lower an erray by mapping contents to a region of memory |
```
Comment (jayphelps) [inline proposals/interface-types/working-notes/instr.md]: s/erray/array

Comment (fgmccabe) [reply proposals/interface-types/working-notes/instr.md]: thanks

File: proposals/interface-types/working-notes/scenarios/points.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  i32.const #8
+  i32.mul
+  call "malloc"
+  deferred (i32)
+    call "free"
+  end
+  let (local $ptr i32) (result (type $returnCode))
+    local.get $pts
+    array-to-memory #8
+      let (local $pt i32) (local $ptr i32) (local $ix i32)
```
Comment (binji) [inline proposals/interface-types/working-notes/scenarios/points.md]: Should this be `(local $pt (type $point))`?

File: proposals/interface-types/working-notes/scenarios/points.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+The array is modeled as two values: a pointer to its base and the number of
+elements in the array.
+
+The primary pattern with processing arrays is the for-loop pattern, together
+with array indexing; which is modeled by the `array-to-memory` instruction;
+which iterates a code fragment over a range of numbers.
+
+```
+(@interface datatype @point
+   (tuple i32 i32))
```
Comment (binji) [inline proposals/interface-types/working-notes/scenarios/points.md]: `(tuple s32 s32)`?

File: proposals/interface-types/working-notes/scenarios/points.md (hunk 2 above)
Comment (fgmccabe) [reply proposals/interface-types/working-notes/scenarios/points.md]: Yes, you are right. Note that this is subject to change,

**Pull request closed without merging**